src/components/DatePicker/index.ios.js

```diff
@@ -1,163 +1,147 @@
+    const initialValue = useRef(null);
+    const inputRef = useRef(null);
 
-class DatePicker extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isPickerVisible: false,
-            selectedDate: props.value || props.defaultValue ? moment(props.value || props.defaultValue).toDate() : new Date(),
-        };
-
-        this.showPicker = this.showPicker.bind(this);
-        this.reset = this.reset.bind(this);
-        this.selectDate = this.selectDate.bind(this);
-        this.updateLocalDate = this.updateLocalDate.bind(this);
-    }
-
-    showPicker() {
-        this.initialValue = this.state.selectedDate;
+    const showPicker = useCallback(() => {
+        initialValue.current = selectedDate;
 
         // Opens the popover only after the keyboard is hidden to avoid a "blinking" effect where the keyboard was on iOS
         // See https://github.com/Expensify/App/issues/14084 for more context
-        if (!this.props.isKeyboardShown) {
-            this.setState({isPickerVisible: true});
+        if (!isKeyboardShown) {
+            setIsPickerVisible(true);
             return;
         }
+
         const listener = Keyboard.addListener('keyboardDidHide', () => {
-            this.setState({isPickerVisible: true});
+            setIsPickerVisible(true);
             listener.remove();
         });
         Keyboard.dismiss();
-    }
+    }, [isKeyboardShown, selectedDate]);
+
+    useEffect(() => {
+        if (!isFunction(innerRef)) {
+            return;
+        }
+
+        const input = inputRef.current;
+
+        if (input && input.focus && typeof input.focus === 'function') {
+            innerRef({...input, focus: showPicker});
+            return;
+        }
+
+        innerRef(input);
+    }, [innerRef, showPicker]);
 
     /**
      * Reset the date spinner to the initial value
      */
-    reset() {
-        this.setState({selectedDate: this.initialValue});
-    }
+    const reset = () => {
+        setSelectedDate(initialValue.current);
+    };
 
     /**
      * Accept the current spinner changes, close the spinner and propagate the change
-     * to the parent component (props.onInputChange)
+     * to the parent component (onInputChange)
      */
-    selectDate() {
-        this.setState({isPickerVisible: false});
-        const asMoment = moment(this.state.selectedDate, true);
-        this.props.onInputChange(asMoment.format(CONST.DATE.MOMENT_FORMAT_STRING));
-    }
+    const selectDate = () => {
+        setIsPickerVisible(false);
+        const asMoment = moment(selectedDate, true);
+        onInputChange(asMoment.format(CONST.DATE.MOMENT_FORMAT_STRING));
+    };
 
     /**
      * @param {Event} event
-     * @param {Date} selectedDate
+     * @param {Date} date
      */
-    updateLocalDate(event, selectedDate) {
-        this.setState({selectedDate});
-    }
+    const updateLocalDate = (event, date) => {
+        setSelectedDate(date);
+    };
```

Comment:
Seems like we could remove this function and call `setSelectedDate` directly?

Reply:
I don't think so as we need only a second argument

Reply:
I guess in this case better to have the arrow function as a variable that passed